src/main/java/org/mondemand/Client.java

```diff
@@ -390,6 +406,24 @@ public void removeAllContexts() {
     }
   }
 
+  /**
+   * verifies if a key is a valid mondemand key (a-Z A-Z 0-9 _ - .)
+   *
+   * @param key - the input key
+   * @return true if the key consists of valid characters, false if otherwise
+   *         or key is empty.
+   */
+  public static boolean keyIsValid(String key) {
+    if(key == null || key.isEmpty()) {
+      return false;
+    }
+    Matcher matcher = keyPattern.matcher(key);
```

Comment:
this is likely going to get called a lot.  I bet there's a more efficient way of doing this check than a regex.  regex is clean to read, but instantiating a Matcher on every call to this has some underlying allocation overhead that we probably don't want.